src/Storage/StorageManager.php

```diff
@@ -361,4 +368,44 @@ private function isUsageSupported(StorageInterface $adapter, int $usageRequested
 
         return $usageRequested >= $adapter->getUsage();
     }
+
+    /**
+     * @param StorageInterface[] $missingIn
+     */
+    private function hotSynchronize(string $hash, StorageInterface $source, array $missingIn): void
+    {
+        if (empty($missingIn)) {
+            return;
+        }
+        try {
+            $size = $this->getSize($hash);
+            $filteredAdapters = [];
+            foreach ($missingIn as $adapter) {
+                if ($size < $adapter->getHotSynchronizeLimit()) {
+                    $filteredAdapters[] = $adapter;
+                }
+            }
+
+            if (empty($filteredAdapters)) {
+                return;
+            }
+
+            foreach ($filteredAdapters as $adapter) {
+                $adapter->initUpload($hash, $size, 'hotSynchronized', 'application/bin');
+            }
+
+            $stream = $source->read($hash);
+            while (!$stream->eof()) {
+                $chunk = $stream->read(4096);
+                foreach ($filteredAdapters as $adapter) {
+                    $adapter->addChunk($hash, $chunk);
+                }
+            }
+
+            foreach ($filteredAdapters as $adapter) {
+                $adapter->finalizeUpload($hash);
+            }
+        } catch (\Throwable $e) {
+        }
```

Comment:
Empty catch, maybe add logging otherwise we do not see these exceptions

Reply:
Fixed